src/installer.ts

```diff
@@ -0,0 +1,201 @@
+ * limitations under the License.
+ */
+
+import * as path from 'path';
+import * as fs from 'fs';
+import mkdirp = require('mkdirp');
+import {platform} from 'os';
+import { Errorable, failed } from './errorable';
+import * as extension from './extension';
+import * as config from './config';
+import * as kamelCli from './kamel';
+import * as shell from './shell';
+import * as vscode from 'vscode';
+import * as kubectlutils from './kubectlutils';
+import * as downloader from './downloader';
+
+const downloadTarball = require('download-tarball');
+
+export const kamel = 'kamel';
+export const kamel_windows = 'kamel.exe';
+export const version = '1.0.0-M1'; //need to retrieve this if possible, but have a default
+
+export function isKamelAvailable() : Promise<boolean> {
+	return new Promise<boolean>( async (resolve, reject) => {
+		let kamelLocal = kamelCli.create();
+		await kamelLocal.invoke('')
+			.then( (rtnValue) => {
+				resolve(true);
+				return;
+			}).catch ( (error) => {
+				console.log(`Apache Camel K CLI (kamel) unavailable: ${error}`);
+				reject(new Error(error));
+				return;
+		});
+	});
+}
+
+export function checkKamelCLIVersion() : Promise<string> {
+	return new Promise<string>( async (resolve, reject) => {
+		let kamelLocal = kamelCli.create();
+		await kamelLocal.invoke('version')
+			.then( (rtnValue) => {
+				const strArray = rtnValue.split(' ');
+				const version = strArray[strArray.length - 1].trim();
+				console.log(`Apache Camel K CLI (kamel) version returned: ${version}`);
+				resolve(version);
+				return;
+			}).catch ( (error) => {
+				console.log(`Apache Camel K CLI (kamel) unavailable: ${error}`);
+				reject(new Error(error));
+				return;
+		});
+	});
+}
+
+export function isKubernetesAvailable(): Promise<boolean> {
+	return new Promise<boolean>( async (resolve, reject) => {
+		const version = await kubectlutils.getKubernetesVersion();
+		if (version) {
+			resolve(true);
+			return;
+		}
+		reject (new Error('No kubectl version found'));
+		return;
+	});
+}
+
+export async function installKamel(context: vscode.ExtensionContext): Promise<Errorable<null>> {
+	await checkKamelCLIVersion().then((currentVersion) => {
+		if (version.toLowerCase() === currentVersion.toLowerCase()) {
+			// no need to install, it's already here
+			extension.shareMessageInMainOutputChannel(`Apache Camel K CLI version ${currentVersion} available`);
+			return { succeeded: true, result: null };
+		}
+	}).catch ( (error) => {
+		console.error(error);
+	});
+
+	const os = platform();
+	const isWindows = (os === 'win32');
+	const binFile = (!isWindows) ? kamel : kamel_windows;
+
+	const installFolder = getInstallFolder(kamel, context);
+	console.log(`Downloading Apache Camel K CLI to ${installFolder}`);
+	mkdirp.sync(installFolder);
+
+	const kamelUrl = `https://github.com/apache/camel-k/releases/download/${version}/camel-k-client-${version}-${os}-64bit.tar.gz`;
```

Comment:
have you tried on Windows already?
I don't think that it is working, because platform() is returning `win32` although the download url contains `windows` https://github.com/apache/camel-k/releases/download/1.0.0-M2/camel-k-client-1.0.0-M2-windows-64bit.tar.gz

Same question for mac, it seems that it is returning `darwin` but `mac is expected `https://github.com/apache/camel-k/releases/download/1.0.0-M2/camel-k-client-1.0.0-M2-mac-64bit.tar.gz

from NodeJS, the constants for os used:
```
    type Platform = 'aix'
        | 'android'
        | 'darwin'
        | 'freebsd'
        | 'linux'
        | 'openbsd'
        | 'sunos'
        | 'win32'
        | 'cygwin';
```

Reply:
Will check

Reply:
Fixed